gui/src/3D/controls/slider3D.ts

```diff
@@ -0,0 +1,193 @@
+import { Observable } from "babylonjs/Misc/observable";
+import { Vector3 } from "babylonjs/Maths/math.vector";
+import { TransformNode } from "babylonjs/Meshes/transformNode";
+import { Scene } from "babylonjs/scene";
+import { Mesh } from "babylonjs/Meshes/mesh";
+import { Control3D } from "./control3D";
+import { MeshBuilder } from "babylonjs/Meshes/meshBuilder";
+import { PointerDragBehavior } from "babylonjs/Behaviors/Meshes/pointerDragBehavior";
+import { Material } from "babylonjs/Materials/material";
+import { StandardMaterial } from "babylonjs/Materials/standardMaterial";
+import { Color3 } from "babylonjs/Maths/math.color";
+import { AbstractMesh } from "babylonjs/Meshes/index";
+
+export interface ISlider3DOptions {
+    minimum?: number,
+    maximum?: number,
+    value?: number
+};
+
+/**
+ * Class used to create a slider in 3D
+ */
+ export class Slider3D extends Control3D {
+    private _sliderBarMaterial: Material;
+    private _sliderThumbMaterial: Material;
+    private _sliderThumb: Mesh;
+    private _sliderBar: Mesh;
+
+    private _minimum: number;
+    private _maximum: number;
+    private _value: number;
+
+    /** Observable raised when the sldier value changes */
+    public onValueChangedObservable = new Observable<number>();
+
+    /**
+     * Creates a new slider
+     * @param name defines the control name
+     */
+     constructor(name?: string, options?: ISlider3DOptions) {
+        super(name);
+
+        this._minimum =  options?.minimum ? options.minimum : 0;
+        this._maximum = options?.maximum ? options.maximum : 100;

```

Comment:
So by default, there are 100 different values the slider can be set to. Do we want that many as a default?

Reply:
I feel like I have seen value intervals added here as well for other slider controls. we probably want this so that we can support high precision as well as integer value scenarios

Reply:
I took this from 2D slider control, which allows the slider be set to 100 values by default. We don't have to use 100 values. Please let me know your recommendation.  Thanks